internal/plugin.go

```diff
@@ -513,6 +536,43 @@ func (p *Plugin) applyDefaults(unmarshaledConfig map[string]interface{}) {
 			policy.Disabled = p.PolicyDefaults.Disabled
 		}
 
+		ignorePending, ignorePendingIsSet := getPolicyBool(unmarshaledConfig, i, "ignorePending")
+		if ignorePendingIsSet {
+			policy.IgnorePending = ignorePending
+		} else {
+			policy.IgnorePending = p.PolicyDefaults.IgnorePending
+		}
+
+		for i, dep := range policy.Dependencies {
+			if dep.Namespace == "" {
+				policy.Dependencies[i].Namespace = p.PolicyDefaults.Namespace
+			}
+
+			if dep.Compliance == "" {
+				policy.Dependencies[i].Compliance = "Compliant"
+			}
+
+			if dep.Kind == "" {
+				policy.Dependencies[i].Kind = policyKind
+			}
+
+			if dep.APIVersion == "" {
+				policy.Dependencies[i].APIVersion = policyAPIVersion
+			}
+		}
+
+		policy.Dependencies = append(policy.Dependencies, p.PolicyDefaults.Dependencies...)
```

Comment:
Everywhere else in the code, if you set it at a lower level, it replaces the value and doesn't act as a union. I would like the behavior to be consistent.

Reply:
This would also need to account if I explicitly set an empty array for `dependencies` at the policy level so the policy default is not inherited.

Reply:
Consistency would be good...

If the user sets an empty array for `dependencies` in a policy, *and* has `orderViaDependencies` set, how do you think those should interact? Right now, everything is merged so it's unambiguous (but inconsistent with other settings). I think I'd like the policy to still have the `order` dependency set - if I wanted a policy without that, I'd put it in a separate generator file.

Reply:
So it sounds like setting `dependencies` and setting `orderViaDependencies` should be mutually exclusive and throw an error? I think that's how those sorts of conflicts are handled elsewhere. I guess it's either that or have the ordering skip over policies that have `dependencies` specified in the ordering logic and continue with the next policy in the list?

Reply:
That's a good question. I lean towards `dependencies` and `orderViaDependencies` has being a union, but I would be okay with erroring if both are set.